test/integration/multicluster/install_test.go

```diff
@@ -77,47 +135,234 @@ func TestMulticlusterResourcesPostInstall(t *testing.T) {
 	}
 }
+
+//////////////////////
+///   CERT UTILS   ///
+//////////////////////
+
+func createMulticlusterCertificates() (multiclusterCerts, error) {
+	caKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return multiclusterCerts{}, err
+	}
+
+	var serialNumber int64 = 1
+	caTemplate := createCertificateTemplate(true, "root.linkerd.cluster.local", big.NewInt(serialNumber))
+	// Create self-signed CA. Pass in its own pub key and its own private key
+	caDerBytes, err := x509.CreateCertificate(rand.Reader, &caTemplate, &caTemplate, &caKey.PublicKey, caKey)
+	if err != nil {
+		return multiclusterCerts{}, err
+	}
+
+	// Increment serial number to generate next certificate (issuer)
+	serialNumber++
+	issuerDerBytes, issuerECKey, err := createIssuerCertificate(serialNumber, &caTemplate, caKey)
+	if err != nil {
+		return multiclusterCerts{}, err
+	}
+
+	// Convert keypairs to DER encoding. We don't care about the CA key so we
+	// only encode (to export) the issuer keys
+	issuerDerKey, err := x509.MarshalECPrivateKey(issuerECKey)
+	if err != nil {
+		return multiclusterCerts{}, err
+	}
+
+	// Finally, get strings from der blocks
+	// we don't care about CA's private key, it won't be written to a file
+	ca, _, err := tryDerToPem(caDerBytes, []byte{})
+	if err != nil {
+		return multiclusterCerts{}, err
+	}
+
+	issuer, issuerKey, err := tryDerToPem(issuerDerBytes, issuerDerKey)
+	if err != nil {
+		return multiclusterCerts{}, err
+	}
+
+	return multiclusterCerts{
+		ca:         ca,
+		issuerCert: issuer,
+		issuerKey:  issuerKey,
+	}, nil
+}
+
+// createCertificateTemplate will bootstrap a certificate based on the arguments
+// passed in, with a validty of 24h
+func createCertificateTemplate(isCA bool, subjectCommonName string, serialNumber *big.Int) x509.Certificate {
+	crt := x509.Certificate{
+		SerialNumber:          serialNumber,
+		Subject:               pkix.Name{CommonName: subjectCommonName},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().Add(time.Hour * 24),
+		KeyUsage:              x509.KeyUsageCertSign | x509.KeyUsageCRLSign,
+		BasicConstraintsValid: true,
+		MaxPathLen:            0,
+		IsCA:                  true,
+	}
+	if !isCA {
+		crt.KeyUsage = x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign | x509.KeyUsageCRLSign
+	}
+	return crt
+}
+
+// createIssuerCertificate accepts a serial number, a CA template and the CA's
+// key and it creates and signs an intermediate certificate. The function
+// returns the certificate in DER encoding along with its keypair
+func createIssuerCertificate(serialNumber int64, caTemplate *x509.Certificate, caKey *ecdsa.PrivateKey) ([]byte, *ecdsa.PrivateKey, error) {
+	// Generate keypair first
+	key, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return []byte{}, nil, err
+	}
+
+	// Create issuer template
+	template := createCertificateTemplate(false, "identity.linkerd.cluster.local", big.NewInt(serialNumber))
```

Comment:
Yeah, I think the way I organised this is a bit confusing. At first `isCA` was used for the `IsCA` field (like you noted in the review). After some trial and error, I noticed both have to be "CAs" since there's no concept of an "intermediate-ca" in code.

The only thing we do different for issuer is add `DigitalSignature` to the key usage. I think perhaps it'd be more clear if we get rid of the arg here.